pkg/cloud/gcp/actuators/machine/reconciler.go

```diff
@@ -663,15 +663,38 @@ func (r *Reconciler) ensureInstanceGroup(instanceGroupName string) error {
 	return nil
 }
 
+// ensureCorrectNetworkAndSubnetName makes sure that we are working with the correct network and subnet.
+// In a case of VPC, we have to look whether the expect name of the network and subnet resource
+// matches the one, that is actually up in the cluster.
+func (r *Reconciler) ensureCorrectNetworkAndSubnetName() (string, string, error) {
+	actualNetworkName := fmt.Sprintf("%s-network", r.machine.Labels[machinev1.MachineClusterIDLabel])
+	actualSubnetworkName := fmt.Sprintf("%s-%s-subnet", r.machine.Labels[machinev1.MachineClusterIDLabel], r.machineScope.machine.ObjectMeta.Labels[openshiftMachineRoleLabel])
+
+	for _, network := range r.providerSpec.NetworkInterfaces {
+		if network.Network == actualNetworkName && network.Subnetwork == actualSubnetworkName {
+			return actualNetworkName, actualSubnetworkName, nil
+		}
+	}
+	actualNetworkName = r.providerSpec.NetworkInterfaces[0].Network
+	actualSubnetworkName = r.providerSpec.NetworkInterfaces[0].Subnetwork
+
+	return actualNetworkName, actualSubnetworkName, nil
+}
+
 // registerNewInstanceGroup registers an instance group when there is an instance
 // that is using that unkown instance group.
 func (r *Reconciler) registerNewInstanceGroup() error {
-	_, err := r.computeService.InstanceGroupInsert(r.projectID, r.providerSpec.Zone, &compute.InstanceGroup{
+	actualNetworkName, actualSubnetworkName, err := r.ensureCorrectNetworkAndSubnetName()
+	if err != nil {
+		return fmt.Errorf("failed to ensure the correct network name: %v", err)
```

Comment:
Nit, use %w for errors
```suggestion
		return fmt.Errorf("failed to ensure the correct network name: %w", err)
```